contract/src/lib.rs

```diff
@@ -388,6 +402,50 @@ impl Contract {
             false
         }
     }
+
+    // Current GC implementation is doing full travers of the blocks starting from the main chain
+    // tip. We can optimize this, by storing the height of a current genesis block in our state.
+    pub fn run_mainchain_gc(&mut self, batch_size: u8) {
```

Comment:
It is better to declare batch_size as u64 instead of casting it.

Reply:
done